### ISSUE TITLE
Bpytop high CPU usage [BUG]

### ISSUE THREAD

**Issue by alexedurante** [aristocratos/bpytop#57]
**Describe the bug**

bpytop creates a python 3 process that consumes 90% of a single CPU core or 90% combined across 2 CPU cores

**To Reproduce**

Start bpytop as normal

**Expected behavior**

bpytop to not run using a single core so heavily.

**Screenshots**

![image](https://user-images.githubusercontent.com/22942706/89914897-b03b3f80-dbc3-11ea-966b-a08e5bd14e48.png)

**Info (please complete the following information):**
 - bpytop version: 1.0.10
 - psutil version: `5.7.2
 - (Linux) Linux distribution and version: Arch w/ kernel 5.7.12-arch1-1 x86_64
 - Terminal used: Kitty
 - Font used: MesloLGS NF 10pt
 - Python version: 3.8.5

**Additional context**

contents of `~/.config/bpytop/error.log`

(try running bpytop with `--debug` flag if error.log is empty)

error.log was empty. When running with debug flag no issues show
```
11/08/20 (11:19:41) | INFO: New instance of bpytop version 1.0.10 started with pid 53308
11/08/20 (11:19:41) | INFO: Loglevel set to WARNING
11/08/20 (11:19:41) | DEBUG: Using psutil version 5.7.2
11/08/20 (11:19:41) | DEBUG: CMD: /usr/bin/bpytop --debug
11/08/20 (11:19:46) | DEBUG: Collect and draw completed in 2.244664 seconds
11/08/20 (11:19:48) | DEBUG: Init completed in 6.378906 seconds
```

**Comment by aristocratos** [comment]
Well it's using about 30x more cpu than it's supposed too.

What install method did you use?

Since the error.log is empty would you mind doing some tests?
1. Try running as another user (a new clean user preferably)
2. Try running in another terminal

That could maybe tell something if result is different.

**Comment by alexedurante** [comment]
Install method was the AUR
https://aur.archlinux.org/packages/bpytop/

As for your tests, I just tried running in Konsole and am seeing the same issue there
I also tried a fresh install of Tilix and saw the issue there as well

I also just tested on a brand new user and I am not seeing the issue there. Opened all the same applications I would normally have open and python3 behaved like expected

**Comment by aristocratos** [comment]
> I also just tested on a brand new user and I am not seeing the issue there.

Look through your dot files and see if there's anything that might change how terminal programs behave, or possibly if there anything related to python3.

You haven't experienced similar issues with other terminal programs?

**Comment by alexedurante** [comment]
No other terminal programs seem effected, even extremely similar ones. Bashtop works great, gotop works great, bpytop goes after the CPU. Seems to definitely be something relating to python3. Do you know if there is a way to validate if python 3 is configured correctly?

I went through my dot files and didnt see anything effecting python3. However later today if I have more time I will try overwriting some of my existing dot files with those from the new user to see if I can pick out which file at least might be causing the issue.

**Comment by aristocratos** [comment]
You could try the following to rule out if it has to do with psutil being messed up:

`git clone https://github.com/giampaolo/psutil`
`cd psutil`
`python3 scripts/internal/print_api_speed.py`

And then compare the speeds you get to this:
``` text
SYSTEM APIS               SECONDS
----------------------------------
cpu_count                 0.000013
disk_usage                0.000017
sensors_battery           0.000023
cpu_times                 0.000033
cpu_stats                 0.000040
cpu_percent               0.000040
cpu_times_percent         0.000041
boot_time                 0.000041
pid_exists                0.000058
virtual_memory            0.000073
users                     0.000092
swap_memory               0.000097
net_io_counters           0.000124
net_if_stats              0.000154
net_if_addrs              0.000182
disk_partitions           0.000286
disk_io_counters          0.000432
sensors_fans              0.000594
pids                      0.000623
cpu_count (cores)         0.000715
sensors_temperatures      0.004284
process_iter (all)        0.014630
net_connections           0.017292
cpu_freq                  0.078729

PROCESS APIS              SECONDS
----------------------------------
create_time               0.000001
nice                      0.000005
cwd                       0.000009
cpu_affinity              0.000010
ionice                    0.000011
exe                       0.000012
num_fds                   0.000015
memory_info               0.000023
num_threads               0.000023
status                    0.000023
cpu_times                 0.000025
ppid                      0.000027
io_counters               0.000029
num_ctx_switches          0.000031
cpu_num                   0.000031
name                      0.000031
uids                      0.000034
cpu_percent               0.000037
is_running                0.000038
gids                      0.000038
memory_percent            0.000044
open_files                0.000051
threads                   0.000052
cmdline                   0.000052
connections               0.000069
environ                   0.000091
username                  0.000185
terminal                  0.000728
memory_full_info          0.000800
memory_maps               0.002211
children                  0.006912
```

**Comment by alexedurante** [comment]
Here is my output 

```
SYSTEM APIS               SECONDS
----------------------------------
disk_usage                0.000015
cpu_count                 0.000033
boot_time                 0.000043
users                     0.000054
cpu_times                 0.000103
disk_io_counters          0.000122
cpu_times_percent         0.000124
pid_exists                0.000126
cpu_percent               0.000130
virtual_memory            0.000186
disk_partitions           0.000189
cpu_stats                 0.000206
swap_memory               0.000254
pids                      0.001187
sensors_fans              0.001763
cpu_count (cores)         0.002060
net_if_addrs              0.002061
net_if_stats              0.002153
net_io_counters           0.002300
sensors_battery           0.002550
sensors_temperatures      0.010343
process_iter (all)        0.027222
net_connections           0.048213
cpu_freq                  0.199538

PROCESS APIS              SECONDS
----------------------------------
create_time               0.000001
nice                      0.000005
cpu_affinity              0.000009
cwd                       0.000011
exe                       0.000012
ionice                    0.000012
num_fds                   0.000017
status                    0.000023
num_threads               0.000025
ppid                      0.000028
memory_info               0.000029
cpu_times                 0.000029
io_counters               0.000031
num_ctx_switches          0.000031
uids                      0.000033
name                      0.000033
cpu_num                   0.000034
gids                      0.000038
is_running                0.000039
cpu_percent               0.000039
memory_percent            0.000041
cmdline                   0.000046
threads                   0.000049
open_files                0.000052
connections               0.000060
environ                   0.000083
username                  0.000165
terminal                  0.000609
memory_full_info          0.001130
memory_maps               0.003859
children                  0.009056
```
Nothing seems too out of wack outside of CPU_Freq at almost 0.2 seconds

**Comment by aristocratos** [comment]
The values seems fine. I'm just curious about this line from your error.log

> 11/08/20 (11:19:46) | DEBUG: Collect and draw completed in 2.244664 seconds

That time should be around 0.1 seconds. But no idea why it's taking so long and what could possibly effect that externally except psutil.

**Comment by alexedurante** [comment]
Does bashtop do something similar for that step? I could run them both concurrently in different terminal windows for an hour or so and see if any large differences show when I pull the error logs for each?

If it helps, when I run bashtop the psutil option is "on" so they should both backend to the same place if I understand correctly

**Comment by aristocratos** [comment]
Bashtop would have high cpu usage also if it was effected by the same error.

The reason for the high cpu usage is that the time it takes to collect all needed data takes longer than your update time at 2000ms. So it's constantly collecting without pause.

Can you post your `bpytop.conf` file, just to make sure it's not a combination of settings I haven't thought of that's causing it.

**Comment by alexedurante** [comment]
Here is the bpytop.conf

```
#? Config file for bpytop v. 1.0.10

#* Color theme, looks for a .theme file in "/usr/[local/]share/bpytop/themes" and "~/.config/bpytop/themes", "Default" for builtin default theme.
#* Prefix name by a plus sign (+) for a theme located in user themes folder, i.e. color_theme="+monokai"
color_theme="Default"

#* Update time in milliseconds, increases automatically if set below internal loops processing time, recommended 2000 ms or above for better sample times for graphs.
update_ms=2000

#* Processes sorting, "pid" "program" "arguments" "threads" "user" "memory" "cpu lazy" "cpu responsive",
#* "cpu lazy" updates top process over time, "cpu responsive" updates top process directly.
proc_sorting="cpu responsive"

#* Reverse sorting order, True or False.
proc_reversed=False

#* Show processes as a tree
proc_tree=False

#* Use the cpu graph colors in the process list.
proc_colors=True

#* Use a darkening gradient in the process list.
proc_gradient=True

#* If process cpu usage should be of the core it's running on or usage of the total available cpu power.
proc_per_core=True

#* Show process memory as bytes instead of percent
proc_mem_bytes=False

#* Check cpu temperature, needs "vcgencmd" on Raspberry Pi and "osx-cpu-temp" on MacOS X.
check_temp=True

#* Draw a clock at top of screen, formatting according to strftime, empty string to disable.
draw_clock="%X"

#* Update main ui in background when menus are showing, set this to false if the menus is flickering too much for comfort.
background_update=True

#* Custom cpu model name, empty string to disable.
custom_cpu_name="i7-8700k"

#* Optional filter for shown disks, should be last folder in path of a mountpoint, "root" replaces "/", separate multiple values with comma.
#* Begin line with "exclude=" to change to exclude filter, oterwise defaults to "most include" filter. Example: disks_filter="exclude=boot, home"
disks_filter=""

#* Show graphs instead of meters for memory values.
mem_graphs=True

#* If swap memory should be shown in memory box.
show_swap=True

#* Show swap as a disk, ignores show_swap value above, inserts itself after first disk.
swap_disk=True

#* If mem box should be split to also show disks info.
show_disks=True

#* Minimum value the network graphs scales down to, default "10K" = 10 KibiBytes, possible units "K" (KiB), "M" (MiB), "G" (GiB), no unit for bytes.
net_download_min="10K"
net_upload_min="10K"

#* Start in network graphs auto rescaling mode, ignores any values set above and rescale down to default value "10K".
net_auto_min=True

#* Show init screen at startup, the init screen is purely cosmetical
show_init=True

#* Enable check for new version from github.com/aristocratos/bpytop at start.
update_check=True

#* Enable start in mini mode, can be toggled with shift+m at any time.
mini_mode=False

#* Set loglevel for "~/.config/bpytop/error.log" levels are: "ERROR" "WARNING" "INFO" "DEBUG".
#* The level set includes all lower levels, i.e. "DEBUG" will show all logging info.
log_level=WARNING

```

**Comment by aristocratos** [comment]
I'm getting the same issue with that config file.

Gonna investigate.

**Comment by aristocratos** [comment]
I found it, had forgotten a parenthesis when showing memory in percent instead of bytes.

Have no idea how I didn't catch this before...

**Comment by aristocratos** [comment]
Fixed in v1.0.11

**Comment by alexedurante** [comment]
Just downloaded the update and this issue is resolved! I will now close this issue
